tests/PullRequest.py

```diff
@@ -402,3 +407,70 @@ def testUpdateBranch(self):
             self.pull.update_branch("addaebea821105cf6600441f05ff2b413ab21a36")
         )
         self.assertTrue(self.pull.update_branch())
+
+    def testDeleteOnMerge(self):
+        self.assertTrue(
+            self.delete_restore_repo.get_branch(self.delete_restore_pull.head.ref)
+        )
+        self.assertFalse(self.delete_restore_pull.is_merged())
+        status = self.delete_restore_pull.merge(deletebranch=True)
+        self.assertTrue(status.merged)
+        self.assertTrue(self.delete_restore_pull.is_merged())
+        with self.assertRaises(GithubException) as raisedexp:
```

Comment:
with self.assertRaises(github.GithubException) ...   which also means you don't need the import at the top